gaia_tools/xmatch/__init__.py

```diff
@@ -181,9 +185,33 @@ def cds(cat,xcat='vizier:I/345/gaia2',maxdist=2,colRA='RA',colDec='DEC',
     return (ma,mai)
 
 def cds_load(filename):
-    return numpy.genfromtxt(filename,delimiter=',',skip_header=0,
-                            filling_values=-9999.99,names=True,
-                            dtype='float128')
+    if WIN32:
+        # windows do not have float128, but source_id is double
+        # get around this by squeezing precision from int64 on source_id as source_id is always integer anyway
+        # first read everything as fp64 and then convert source_id to int64 will keep its precision
+        with open(filename, 'r'):
+            # only read the first row max to reduce workload to just get the column name
+            data = numpy.genfromtxt(filename, delimiter=',', skip_header=0,
+                                    filling_values=-9999.99, names=True, max_rows=1,
+                                    dtype='float64')
+            to_list = list(data.dtype.names)
+            # construct a list where everything is fp64 except 'source_id' being int64
+            dtype_list = [('{}'.format(i), numpy.float64) for i in to_list]
+            # find the index of source_id in list
```

Comment:
Can't you just do
```
dtype_list[dtype_list.index(('source_id', numpy.float64))]= ('source_id', numpy.uint64)
```
? (or similar, didn't test)

Also, is it necessary to do the ``with open(filename,'r')``?

Reply:
yes, did not know that feature. The reason I use `with` was because I thought we need to read the whole file twice and probably be problematic if the file is way larger than RC-DR2 cross-matched. Now I defined `max_rows=1` in the first read. I guess it is fine so I removed that now.